icaruscode/Decode/ChannelMapping/ChannelMapPostGres_tool.cc

```diff
@@ -84,13 +99,53 @@ class ChannelMapPostGres : virtual public IChannelMapping
   virtual int BuildSideCRTCalibrationMap(SideCRTChannelToCalibrationMap&) const override;
 
 private:
+
+    /// Record of all relevant table names.
+    struct TableNames_t {
+      std::string TPCfragmentMap;
+      std::string TPCreadoutBoardMap;
+      std::string PMTfragmentMap;
+      std::string CRTsideMap;
+      std::string CRTtopMap;
+      std::string CRTsideCalibrationMap;
+    };
+
     // Recover data from postgres database
     int GetDataset(const std::string&, const std::string&, const std::string&, Dataset&) const;
     int GetCRTCaldata(const std::string&, const std::string&, Dataset&) const;
     uint32_t fNothing;     //< Nothing
 
+    /// The set of tables being served. Chosen by `SelectRun()`.
+    TableNames_t const* fCurrentTable = nullptr;
+
+    /// The list of names of tables.
+    static std::array<TableNames_t, RunPeriods::NPeriods> const TableNameSets;
+
+};
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+std::array<ChannelMapPostGres::TableNames_t, RunPeriods::NPeriods> const
+ChannelMapPostGres::TableNameSets{
+  TableNames_t{
+      "icarus_hw_readoutboard"    // TPCfragmentMap
+    , "icarus_hardware_prd"       // TPCreadoutBoardMap
```

Comment:
There is a copy and paste error here: `"icarus_hardware_prd"` is constant for all queries (it's the `dbname`). The TPC readout board map table should be simply `"daq_channels"` as below.

Reply:
I may have fixed this an even number of times, while odd is required. :facepalm: